reefperf/cloud_node.py

```diff
@@ -1,14 +1,30 @@
-from abc import ABCMeta, abstractmethod
+import subprocess
+from abc import abstractmethod
+from logfury.v0_1 import DefaultTraceAbstractMeta
 
 from reefperf.node_connection import ParamikoConnection
+from reefperf import utils
 
 
-class CloudNode(object, metaclass=ABCMeta):
+class CloudNode(object, metaclass=DefaultTraceAbstractMeta):
     @property
     @abstractmethod
     def connection(self):
         pass
 
+    @property
+    @abstractmethod
+    def deploy_command(self):
+        pass
+
+    @property
+    @abstractmethod
+    def ssh_data(self):
+        pass
+
+    def deploy(self):
+        completed = subprocess.run([f"{self._deploy_command} {self.ssh_data}"], shell=True, stdout=subprocess.PIPE)
```

Comment:
that is not how one safely escapes data when calling a subprocess. Please consider the username which has a space in it, hostile config provided by a customer which has a `"; rm -rf /; "` etc